pkg/piecestore/psservice/config.go

```diff
@@ -34,24 +32,24 @@ func (c Config) Run(ctx context.Context, server *provider.Provider) (
 	err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	ttlPath := filepath.Join(c.Path, "ttl.db")
-	piecePath := filepath.Join(c.Path, "data")
+	serverConf := psserver.Config{
+		PieceStoreDir: c.Path,
+	}
 
-	ttldb, err := ttl.NewTTL(ttlPath)
+	s, err := psserver.Initialize(serverConf)
 	if err != nil {
 		return err
 	}
 	// TODO(jt): defer ttldb.Close()
 
 	// TODO(jt): server.Server constructor
-	s := &psserver.Server{PieceStoreDir: piecePath, DB: ttldb}
 	// TODO(jt): defer s.Close()
 
 	pspb.RegisterPieceStoreRoutesServer(server.GRPC(), s)
 
 	go func() {
 		// TODO(jt): why isn't the piecestore server doing this?
-		log.Fatal(s.DB.DBCleanup(piecePath))
+		log.Fatal(s.Stop(ctx))
```

Comment:
wait, s.Stop()? in a goroutine? right before running the server?

it does look like you can clean up a bunch of my TODOs, but i don't think this is right